microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/settings/capabilities/ServerCapabilitiesConstants.java

```diff
@@ -44,8 +44,12 @@ private ServerCapabilitiesConstants() {
 	public static final String CODE_ACTION_ID = UUID.randomUUID().toString();
 	public static final String CODE_LENS_ID = UUID.randomUUID().toString();
 
-	public static final CompletionOptions DEFAULT_COMPLETION_OPTIONS = new CompletionOptions(false, Arrays.asList(".",
-			"%", "=" /* triggered characters for properties file */ , "@" /* triggered characters for java snippets annotation */));
+	public static final CompletionOptions DEFAULT_COMPLETION_OPTIONS = new CompletionOptions(false,
+			Arrays.asList(".", "%", "=" /* triggered characters for properties file */ ,
+					"@" /* triggered characters for java snippets annotation */));
 
 	public static final CodeLensOptions DEFAULT_CODELENS_OPTIONS = new CodeLensOptions();
+
+	public static final FormattingRegisterOptions DEFAULT_FORMATTING_OPTIONS = new FormattingRegisterOptions(
+			"microprofile-properties", "quarkus-properties");
```

Comment:
"quarkus-properties" should be contributed by the quarkus extension

Reply:
the quarkus extension will also potentially register qute-* languages

Reply:
@fbricon you mean that client should send the supported language (file properties) and server should have a client capability for that?

Reply:
> the quarkus extension will also potentially register qute-* languages

No I don't think so, because for Qute it should be better to have a new language server for Qute

Reply:
the quarkus-ls extension jar should register the quarkus specific stuff, it has nothing to do with the client.

For a Qute-LS, we're not there yet anyway.

Reply:
IMHO I think it's the client which must specify the language IDs for properties file. For instance in Eclipse IDE we have content type (which is different from quarkus-properties). In otherwise I think it's the role of the client to specify which language IDs are for properties file.

Reply:
@fbricon I changed my strategy and I'm using pattern instead of language which is more generic:

```
"registerOptions": {
	"documentSelector": [
		{ "pattern": "**/*.properties" }
	]
}
```

Reply:
that's too broad. Not only do you also target spring boot properties, but also any other *.properties file

Reply:
Ok I'm using language now. For adding quarkus-properties as language document filter I do that with Java SPI in Quarkus LS.